test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PrngPrecompileSuite.java

```diff
@@ -0,0 +1,230 @@
+                                .logged()
+                                .saveTxnRecordToRegistry(largeInputCall),
+                        withOpContext(
+                                (spec, ignore) -> {
+                                    final var gasUsed =
+                                            spec.registry()
+                                                    .getTransactionRecord(largeInputCall)
+                                                    .getContractCallResult()
+                                                    .getGasUsed();
+                                    assertEquals(394244, gasUsed);
+                                }));
+    }
+
+    private HapiApiSpec nonSupportedAbiCallGracefullyFails() {
+        final var prng = THE_GRACEFULLY_FAILING_PRNG_CONTRACT;
+        final var failedCall = "failedCall";
+        return defaultHapiSpec("nonSupportedAbiCallGracefullyFails")
+                .given(cryptoCreate(bob), uploadInitCode(prng), contractCreate(prng))
+                .when(
+                        sourcing(
+                                () ->
+                                        contractCall(prng, "performNonExistingServiceFunctionCall")
+                                                .gas(GAS_TO_OFFER)
+                                                .payingWith(bob)
+                                                .via(failedCall)
+                                                .hasKnownStatus(CONTRACT_REVERT_EXECUTED)
+                                                .logged()))
+                .then(
+                        getTxnRecord(failedCall)
+                                .andAllChildRecords()
+                                .logged()
+                                .saveTxnRecordToRegistry(failedCall),
+                        withOpContext(
+                                (spec, ignore) -> {
+                                    final var gasUsed =
+                                            spec.registry()
+                                                    .getTransactionRecord(failedCall)
+                                                    .getContractCallResult()
+                                                    .getGasUsed();
+                                    assertEquals(394209, gasUsed);
+                                }));
+    }
+
+    private HapiApiSpec functionCallWithLessThanFourBytesFailsGracefully() {
+        final var lessThan4Bytes = "lessThan4Bytes";
+        return defaultHapiSpec("functionCallWithLessThanFourBytesFailsGracefully")
+                .given(
+                        cryptoCreate(bob),
+                        uploadInitCode(THE_GRACEFULLY_FAILING_PRNG_CONTRACT),
+                        contractCreate(THE_GRACEFULLY_FAILING_PRNG_CONTRACT))
+                .when(
+                        sourcing(
+                                () ->
+                                        contractCall(
```

Comment:
The BDD code will transalte this into a 4 byte call.  we need the `withExplicitParams` method to about 3 bytes.